### PR TITLE
Changelog for v0.50.0

### PR DESCRIPTION
**Pull request by kkaefer**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,46 @@
+## 0.50.0
+(Release date TBD)
+
+### ✨ Features and improvements
+* 🎉 Add Custom Layers that can be rendered into with user-provided WebGL code ([#7039](https://github.com/mapbox/mapbox-gl-js/pull/7039))
+* Add WebGL face culling for increased performance ([#7178](https://github.com/mapbox/mapbox-gl-js/pull/7178))
+* Improve speed of expression evaluation ([#7334](https://github.com/mapbox/mapbox-gl-js/pull/7334))
+* Automatically coercce to string for `concat` expresion and `text-field` property ([#6190](https://github.com/mapbox/mapbox-gl-js/issues/6190), fixed by [#7280](https://github.com/mapbox/mapbox-gl-js/pull/7280))
```
Comment (ryanhamley) [inline CHANGELOG.md]: `coercce` -> `coerce`

Comment (ryanhamley) [reply CHANGELOG.md]: `expresion` too

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,46 @@
+## 0.50.0
```
Comment (mollymerp) [inline CHANGELOG.md]: In the past we've put the beta designation in the Changelog and removed it as part of the final release process, but I don't have a strong opinion about whether we should keep doing that or not.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,10 +1,46 @@
...
+	* Allow calling `to-color` on values that are already of type `Color` ([#7260](https://github.com/mapbox/mapbox-gl-js/pull/7260))
+	* Fix `to-array` for empty arrays (([#7261](https://github.com/mapbox/mapbox-gl-js/pull/7261)))
+	* Fix identity functions for `text-field` when using formatted text ([#7351](https://github.com/mapbox/mapbox-gl-js/pull/7351))
+	* Fix coercion of `null` to `0` in `to-number` expression ([#7083](https://github.com/mapbox/mapbox-gl-js/issues/7083), fixed by [#7274](https://github.com/mapbox/mapbox-gl-js/pull/7274))
+* **Canvas source**
+	* Fix missing repeats of `CanvasSource` when it crosses the antimeridian ([#7273](https://github.com/mapbox/mapbox-gl-js/pull/7273))
+	* Fix `CanvasSource` not respecting alpha values set on `canvas` element ([#7302](https://github.com/mapbox/mapbox-gl-js/issues/7302), fixed by [#7309](https://github.com/mapbox/mapbox-gl-js/pull/7309))
+* **Rendering**
+	* Fix rendering of fill extrusions with really high heights ([#7292](https://github.com/mapbox/mapbox-gl-js/pull/7292))
+	* Fix reloading of empty tiles ([#7355](https://github.com/mapbox/mapbox-gl-js/pull/7355))
```
Comment (mollymerp) [inline CHANGELOG.md]: Usually we try to include the user-facing bug in the changelog description, which in this case was an issue where map state would never return to `loaded` after certain runtime styling changes when there were errored tiles in the viewport

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,10 +1,46 @@
...
+	* Fix errors when rendering symbol layers without symbols ([#7241](https://github.com/mapbox/mapbox-gl-js/issues/7241), fixed by [#7253](https://github.com/mapbox/mapbox-gl-js/pull/7253))
```
Comment (mollymerp) [inline CHANGELOG.md]: Hmm these link to an IE error due to empty glyph/image atlases – is that correct?

**Pull request merged**